src/onyx/peer/task_lifecycle.clj

```diff
@@ -94,7 +94,9 @@
 (defn group-message [segment catalog task]
   (let [t (find-task-fast catalog task)]
     (if-let [k (:onyx/group-by-key t)]
-      (hash (get segment k))
+      (if (vector? k)
+        (hash (sort ((apply juxt k) segment)))
```

Comment:
This use of juxt will break groupings with non-keyword keys (which is why get was used rather than (k segment)). It'd be nice to keep this property when using multiple keys.

Is there any reason why we're sorting here? The key order will be consistent over the peers, so I think grouping will be consistent without the sort.